### PR TITLE
Fix isVisible

### PR DESCRIPTION
**Pull request by TravisL12**
Get's tricky passing the array of cards.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+    <meta name="theme-color" content="#000000" />
+    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico" />
+    <title>React App</title>
+  </head>
 
-<head>
-	<meta charset="utf-8">
-	<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
-	<meta name="theme-color" content="#000000">
-	<link rel="manifest" href="%PUBLIC_URL%/manifest.json">
```
Comment (TravisL12) [inline public/index.html:8]: This was causing some weird error, just removed this line. Otherwise ignore this file.

File: src/App.js [hunk 2]
```diff
@@ -5,15 +5,13 @@ import BoardCard from "./BoardCard";
...
 import { sideDeck } from "./sideDeck";
-import { is } from "express/lib/request";
 
 export default function App() {
   const [total, setTotal] = useState(0);
   const [deck, setDeck] = useState(initDeck);
   const [board, setBoard] = useState([]);
   const [sDeck, setSDeck] = useState(sideDeck);
   const [sideBoard, setSideBoard] = useState([]);
-  const [isVisible, setIsVisible] = useState(sideDeck[0].isVisible);
```
Comment (TravisL12) [inline src/App.js:16]: Don't need this anymore. Cause the `isVisible` property is on each card.

File: src/App.js [hunk 3]
```diff
@@ -38,7 +36,7 @@ export default function App() {
         for (let i = 0; i < 4; i++) {
           let newSideCard = Math.floor(Math.random() * sDeck.length);
           const removedSideCard = sDeck.splice(newSideCard, 1);
-          setSideBoard((current) => [...current, removedSideCard]);
+          setSideBoard((current) => [...current, removedSideCard].flat());
```
Comment (TravisL12) [inline src/App.js:39]: `sideBoard` was an array of single arrays, using `flat` will make it an array of values.

File: src/SideBoard.jsx [hunk 4]
```diff
@@ -2,25 +2,26 @@ import React from "react";
...
+  setSideBoard,
+}) {
+  return (
+    <div className="sboard">
+      <h3>Side Board</h3>
+      {sideBoard.map((card) => {
+        return (
+          <SideBoardCard
+            key={card.id}
+            card={card}
```
Comment (TravisL12) [inline src/SideBoard.jsx]: Just pass the card, no need to make a prop for every attribute of that object.

File: src/SideBoardCard.jsx [hunk 5]
```diff
@@ -1,22 +1,32 @@
...
+
+  const { id, value, isVisible } = card;
+
+  function useSideCard() {
+    if (total + value < 0) {
+      console.log("Cannot bring total below zero!");
+    } else {
+      setTotal(total + value);
+      setSideBoard((prevDeck) => {
+        const cardsDup = JSON.parse(JSON.stringify(prevDeck)); // little trick to duplicate array
```
Comment (TravisL12) [inline src/SideBoardCard.jsx:16]: We can't manipulate the deck directly, we have to make a copy of it first. I use this little trick of serializing an array (turn into a string) then convert it back to an array (JSON.parse) to get a proper duplicate.

File: src/SideBoardCard.jsx [hunk 6]
```diff
@@ -1,22 +1,32 @@
...
+        const updateCard = cardsDup.find((c) => c.id === id);
```
Comment (TravisL12) [inline src/SideBoardCard.jsx:17]: Find the card we just clicked within the deck of cards, now we can manipulate the `isVisible` state on that one card.

File: src/SideBoard.jsx [hunk 7]
```diff
@@ -2,25 +2,28 @@ import React from "react";
...
+  sideBoard,
+  setTotal,
+  total,
+  setSideBoard,
+}) {
+  return (
+    <div className="sboard">
+      <h3>Side Board</h3>
+      {sideBoard
+        .filter((card) => card.isVisible)
```
Comment (TravisL12) [inline src/SideBoard.jsx:15]: Instead of using CSS to hide/show the cards just filter out any card that isn't visible.

**Review by spgej — APPROVED** [review]

File: src/styles.css [hunk 8]
```diff
@@ -52,8 +52,3 @@ h3 {
   height: 100px;
   background-color: transparent;
 }
-
-.usedCard {
```
Comment (TravisL12) [inline src/styles.css:56]: Now that we are filtering out the used cards we don't need this anymore.

**Pull request merged**